Geometry_Engine/Query/HashArray.cs

```diff
@@ -66,100 +86,173 @@ public static double[] IHashArray(this IGeometry igeometry)
         [Description("The geometry hash of a Curve is obtained by first retrieving any Sub-part of the curve, if present." +
             "The ISubParts() methods is able to return the 'primitive' curves that a curve is composed of. " +
             "The GeometryHashes are then calculated for the individual parts and concatenated.")]
-        private static double[] HashArray(this ICurve curve, double translationFactor)
+        private static double[] HashArray(this ICurve curve, double translationFactor, BaseComparisonConfig comparisonConfig = null, string fullName = null)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(ICurve).IsAssignableFrom(t)) ?? false)
+                return default;
+
+            if (curve == null)
```

Comment:
I would consider putting it as the 1st check to be made in the method to avoid redundant checks against type exceptions. Minor thing, but may have some impact in case of massive workflows with complex configs?

Reply:
Agreed, applied in e360db11d702c8219f104b7448a521e131f849fe